File: src/models/cooler.model.js
const { Schema, model } = require('mongoose');

const coolerSchema = new Schema({
    name: { type: String, required: true },
    brand: { type: String, required: true },
    type: { type: String, enum: ['Air', 'Liquid'], required: true },
    fan_size_mm: { type: Number, required: true },
    noise_level: { type: Number, required: true },
    compatibility: {
        cpu_socket: { type: String, required: true }, // Compatibilidad con el socket del CPU
        case_clearanse: { type: Boolean, required: true } // Verifica si cabe en el gabinete
    },
    url: { type: String, require: true },
    price: { type: Number, required: true}
});

module.exports = model('cooler', coolerSchema);

File: lib/models/cooler.model.dart
class Cooler {
  String name;
  String brand;
  String type;
  int fanSizeMm;
  int noiseLevel;
  Compatibility compatibility;
  String url;
  int price;

  Cooler({
    required this.name,
    required this.brand,
    required this.type,
    required this.fanSizeMm,
    required this.noiseLevel,
    required this.compatibility,
    required this.url,
    required this.price,
  });

  factory Cooler.fromJson(Map<String, dynamic> json) {
    return Cooler(
      name: json['name'],
      brand: json['brand'],
      type: json['type'],
      fanSizeMm: json['fanSizeMm'],
      noiseLevel: json['noiseLevel'],
      compatibility: Compatibility.fromJson(json['compatibility']),
      url: json['url'],
      price: json['price'],
    );
  }
}

class Compatibility {
  String cpuSocket;
  bool caseClearance;

  Compatibility({
    required this.cpuSocket,
    required this.caseClearance,
  });

  factory Compatibility.fromJson(Map<String, dynamic> json) {
    return Compatibility(
      cpuSocket: json['cpu_socket'],
      caseClearance: json['case_clearanse'],
    );
  }
}
